website_snippet_contact_form/data/website_snippet_contact_form.xml

```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<openerp>
+<data>
+
+<template id="validation_error_page">
+    <t t-call="website.layout">
+        <div id="wrap">
+            <div class="oe_structure oe_empty">
+                <div class="container text-center">
+                    <h1 class="mt32">Validation Error!</h1>
+                    <h3>
+                        <span t-esc="error"/>
+                    </h3>
+                    <p>
+                        <a class="btn btn-primary mt16 mb16"
+                           t-att-href="http_referrer">Go to Back</a>
```

Comment:
Add `onclick="history.go(-1); return false"` to make JS-enabled browsers have more chances to keep fields still filled please.